operators/pkg/controller/elasticsearch/version/common.go

```diff
@@ -265,10 +266,13 @@ func NewPod(
 	if objectMeta.Labels == nil {
 		objectMeta.Labels = map[string]string{}
 	}
-	cfg, err := podSpecCtx.Config.Unpack()
+
+	cfg := v1alpha1.DefaultCfg
+	err := podSpecCtx.Config.Unpack(&cfg)
```

Comment:
I wonder if we could do something clever with a `type` around the shared config implementation to get back to a typesafe `Unpack()` specific to Elasticsearch without the need for developers to remember to use the defaulted struct.

```golang

// in: controllers/elasticsearch/settings
package settings
type CanonicalConfig struct {
 commonsettings.CanonicalConfig
}

func(c *CanonicalConfig) Unpack() (v1alpha1.ElasticsearchSettings, error) {
	cfg := v1alpha1.DefaultCfg
	return c.Unpack(cfg)
}
```

Reply:
_Yes, we can._